Hogajama/hogajama-rs/src/main/java/com/gepardec/hogarama/rest/unitmanagement/UnitApiImpl.java

```diff
@@ -39,6 +39,41 @@ public Response create(SecurityContext securityContext, UnitDto unitDto) {
         Unit unit = translator.fromDto(unitDto);
         service.createUnit(unit);
 
-        return new BaseResponse<>(HttpStatus.SC_CREATED).createRestResponse();
+        return new BaseResponse<>(unitDto, HttpStatus.SC_CREATED).createRestResponse();
+    }
+
+    @Override
+    @Transactional
+    public Response update(String id, SecurityContext securityContext, UnitDto unitDto) {
+        LOG.info("Updating unit with id {}.", id);
+        Unit unit = translator.fromDto(unitDto);
+
+        if (id == null || !id.equals(unitDto.getId().toString())) {
+            return new BaseResponse<>(HttpStatus.SC_BAD_REQUEST).createRestResponse();
```

Comment:
Die Frage ist, wozu ich die Id bei einem Update extra mitgeben muss, wenn sie ohnehin gleich der Id im Dto sein muss. Ist vermutlich abgeschrieben und wenn man es ändert sollte man es generell ändern.

Reply:
Außerdem ist die Fehlermeldung nicht aussagekräftig und kein Log. Wenn das wirklich mal passieren sollte, ist es sehr mühsam den Fehler zu analysieren.

Reply:
> Die Frage ist, wozu ich die Id bei einem Update extra mitgeben muss, wenn sie ohnehin gleich der Id im Dto sein muss. Ist vermutlich abgeschrieben und wenn man es ändert sollte man es generell ändern.

Da hast du absolut recht, zurzeit wird aus dem Body die ID hergenommen, obwohl sie als Parameter schon daherkommt. Das ist keine einfache Änderung, dafür gehören Unit Tests geschrieben und geändert und es muss überall, wo diese API verwendet wird, angepasst werden!

Reply:
update in ca2da9ecb4f0bf7e1ea029010c1a75766ea747d5